packages/runtime-core/src/componentProps.ts

```diff
@@ -104,7 +104,8 @@ type InferPropType<T> = [T] extends [null]
             : T
 
 export type ExtractPropTypes<O> = O extends object
-  ? { [K in RequiredKeys<O>]: InferPropType<O[K]> } &
+  ? { [K in keyof O]?: unknown } &
```

Comment:
Can you add a comment explaining why we need this?

Something along the lines of: This is needed to keep the relation between the option prop and the props, allowing to use ctrl+click to navigate to the prop options. see: #3656

Reply:
Done. :)